### PR TITLE
Adds custom emotes to mutant races [Reworked]

### PR DESCRIPTION
**Pull request by Yellow-Mushroom**
[Feature][QoL][input wanted]

## About the PR 
Note: this is a reworking of my previous PR

Adds the following list of emotes that are custom to each mutant races with sounds for almost all:

-Lizards: Hiss, Sniff, Tongue
Hiss- Hisses (+noise)
Sniff- Sniffs the air
Tongue- Flicks tongue out

-Roaches: Hiss, Chirp, Click
Hiss- Hisses (+noise)
Chirp- Chirps (+noise)
Click- Clicks (+2 noises)

-Cows: Stamp, Snort, Swish
Stamp- Stamps hoove (+noise)
Snort- Snorts
Swish- Swishes their tail (+noise)

-Skeletons: Rattle
Rattle- Rattles (+noise)

Also included is the commands *saurian *roach, *skeleton, and *cow to see each of these emotes, similar to *list/*listtarget
I also hope this encourages others to add more to mutant race emotes, as I believe the more emotes the better

## Why's this needed? 

This is needed as emotes greatly enhance RP and I believe the more emotes the better, that by avoiding ctrl+m typing, players will be more likely to use emotes to increase character RP

## Changelog

```changelog
(u)Yellow-Mushroom
(+)Adds custom emotes for trait mutant races
```

### CODE REVIEW

**Comment by Yellow-Mushroom** [comment]
The issues with the files appear to be fixed

File: code/datums/mutantraces.dm [hunk 1]
```diff
@@ -2019,6 +2064,17 @@
 			if ("milk")
 				if (mob.emote_check(voluntary))
 					.= release_milk()
+			if ("stamp")
+				if (mob.emote_check(voluntary, 50))
+					. = list("<B>[mob]</B> stamps their hoove!", "<i>stamps their hoove!</i>")
```
Comment (ZeWaka) [inline code/datums/mutantraces.dm:2069]:
```suggestion
					. = list("<B>[mob]</B> stamps their hooves!", "<i>stamps their hooves!</i>")
```

File: code/mob/living/carbon/human/procs/emote.dm [hunk 2]
```diff
@@ -454,7 +454,7 @@
 
 			if ("help")
 				src.show_text("To use emotes, simply enter 'me (emote)' in the input bar. Certain emotes can be targeted at other characters - to do this, enter 'me (emote) (name of character)' without the brackets.")
-				src.show_text("For a list of all emotes, use 'me list'. For a list of basic emotes, use 'me listbasic'. For a list of emotes that can be targeted, use 'me listtarget'.")
+				src.show_text("For a list of all emotes, use 'me list'. For a list of basic emotes, use 'me listbasic'. For a list of emotes that can be targeted, use 'me listtarget'. For a list of mutant race unqiue emotes, use 'me saurian', 'me roach', 'me skeleton', or 'me cow'.")
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/emote.dm:457]: not a fan of how this is handled

File: code/datums/mutantraces.dm [hunk 3]
```diff
@@ -1102,6 +1121,15 @@
 	decomposes = FALSE
 	race_mutation = /datum/bioEffect/mutantrace/skeleton
 
+	emote(var/act, var/voluntary)
+		switch(act)
+			if ("rattle")
+				if (mob.emote_check(voluntary, 50))
+					. = list("<B>[mob]</B> rattles", "<i>rattles</i>")
```
Comment (ZeWaka) [inline code/datums/mutantraces.dm:1128]:
```suggestion
					if (!prob(0.1))
						. = list("<B>[mob]</B> rattles", "<i>rattles</i>")
					else
						. = list("<B>[mob]</B> rattles they bones", "<i>rattles they bones</i>")
```

rattle me bones

**Review by ZeWaka — CHANGES_REQUESTED** [review]
notes:

* why do you use 'they' instead of `his_or_her()`? i mean it's grammatically correct so not blocking
* would rather you just display an associated-list style mutantrace emote list or something

File: code/datums/mutantraces.dm (hunk 1 above)
Comment (Yellow-Mushroom) [reply code/datums/mutantraces.dm:2069]: I am very used to using they as opposed to gendered pronouns so it didn't really come into my mind, I can change it if its preferred by others.

To answer the list question, I would prefer that but I couldn't figure out how to handle that and have it look nice

**Comment by Carbadox** [comment]
I think it'd be nice to list some of the new emotes + the emotes help in the minor changes section of the changelog.

File: code/mob/living/carbon/human/procs/emote.dm (hunk 2 above)
Comment (geneva-breaker) [reply code/mob/living/carbon/human/procs/emote.dm:457]: big same

File: code/datums/mutantraces.dm (hunk 1 above)
Comment (DrPainis) [reply code/datums/mutantraces.dm]: change it
singular they is **BIG CRINGE™**

**Pull request closed without merging**